compiler/optimizer/LoopCanonicalizer.cpp

```diff
@@ -1964,7 +2031,9 @@ void TR_LoopCanonicalizer::eliminateRedundantInductionVariablesFromLoop(TR_Regio
       //else
       primaryIncr = (int64_t) primaryInductionVariable->getIncrement();
 
-      //ListIterator<TR_InductionVariable> derivedIt(&derivedInductionVariables);
+      TR_ScratchList<TR::Block> blocksInNaturalLoop(trMemory());
```

Comment:
Unless you are updating this list I think you want to look at using the subnode iterator on the region https://github.com/eclipse/omr/blob/23ce7a13d1b1decbe61bea5ee08e08339b359524/compiler/optimizer/Structure.hpp#L611 to save copying the pointers into the list only to iterate them again.

Reply:
When using the iterator you can say that you know your trivial loop check won't return true if you encounter a nested loop or improper region.